network/dag/publisher.go

```diff
@@ -134,6 +135,21 @@ func (s *replayingDAGPublisher) publish(ctx context.Context) {
 	}
 }
 
+func remove(l *list.List, ref hash.SHA256Hash) {
+	current := l.Front()
+
+	for {
+		if current == nil {
+			return
+		}
+		if current.Value.(hash.SHA256Hash).Equals(ref) {
+			l.Remove(current)
```

Comment:
could there be duplicate resume points? If so, we could loop over the entire list, trying to remove duplicates

Reply:
Very unlikely the transactionAdded callback is called twice. Even so, doesn't hurt to remove the `return` here.